### PR TITLE
Implemented Benchmark for authorship pallet

### PR DESCRIPTION
**Pull request by vjgaur**

### CODE REVIEW

File: pallets/authorship/Cargo.toml [hunk 1]
```diff
@@ -39,12 +41,20 @@ sp-keystore = { git = "https://github.com/dhiway/substrate", branch = "release-v
...
+  "frame-system/runtime-benchmarks",
+  "cord-utilities/runtime-benchmarks",
+]
 std = [
  'codec/std',
  "scale-info/std", 
  'frame-support/std', 
  'frame-system/std',
  'pallet-balances/std',
+ "frame-benchmarking?/std",
```
Comment (roynrishingha) [inline pallets/authorship/Cargo.toml]: Is `?` necessary or a typo?

**Review by roynrishingha — COMMENTED** [review]
Is `?` a typo?

File: pallets/authorship/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+benchmarks! {
+	
+	add {
+		let authors_count: u32 = T::MaxAuthorityProposals::get();
+		let mut authors = Vec::new();
+
+		for i in 0..authors_count {
+			let user: T::AccountId = account("user", i, SEED);
+			authors.push(user);
```
Comment (smohan-dw) [inline pallets/authorship/src/benchmarking.rs]: We might not need this. Using this in the pallet only for events

Comment (smohan-dw) [inline pallets/authorship/src/benchmarking.rs]: Missing the storage insert

File: pallets/authorship/src/benchmarking.rs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		}
+
+		let caller: T::AccountId = account("caller", 0, SEED);;
```
Comment (smohan-dw) [inline pallets/authorship/src/benchmarking.rs]: has extra ;

**Comment by smohan-dw** [comment]
Also, paste the summary of cargo test --all --all-targets --features runtime-benchmarks --no-fail-fast

File: pallets/authorship/Cargo.toml (hunk 1 above)
Comment (vjgaur) [reply pallets/authorship/Cargo.toml]: oh thats typo

Comment (smohan-dw) [reply pallets/authorship/Cargo.toml]: It's not a typo. It's a way to mark the package as optional.

**Review by smohan-dw — APPROVED** [review]

**Pull request merged**